### ISSUE TITLE
build fails on osx

### ISSUE THREAD

**Issue by x37v** [x37v/puredata-rust#1]
shouldn't need to have these symbols defined?

```
xnor-work:~/personal/src/puredata-rust/examples/helloworld$ cargo build --release
   Compiling helloworld v0.1.0 (/Users/alex/personal/src/puredata-rust/examples/helloworld)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-L" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.0.rcgu.o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.1.rcgu.o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.2.rcgu.o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.3.rcgu.o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.4.rcgu.o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.5.rcgu.o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.6.rcgu.o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.helloworld.40a883bh-cgu.7.rcgu.o" "-o" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/libhelloworld.dylib" "-Wl,-exported_symbols_list,/var/folders/xw/3w07fx_x0bj4fhm063rpmrp80000gn/T/rustcnGt6sq/list" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/helloworld.2oqf9yuyw2w5n8tv.rcgu.o" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps" "-L" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/alex/personal/src/puredata-rust/examples/helloworld/target/release/deps/libpuredata_sys-d8be93bc7348947b.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libstd-084d3c32b37d9af9.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libpanic_unwind-8cecbcf5f274d94d.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libbacktrace_sys-b03662eae585a920.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_demangle-61913e6feeeff678.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libunwind-75148971013e38a9.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liblibc-a5c22490cb7dbe67.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/liballoc-d6235fdae9bccd59.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/librustc_std_workspace_core-04d7f10312787f91.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcore-0d676ca0824aedaa.rlib" "/Users/alex/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-ea47f39b75b3f57e.rlib" "-lSystem" "-lresolv" "-lc" "-lm" "-dynamiclib" "-Wl,-dylib"
  = note: Undefined symbols for architecture x86_64:
            "_class_addbang", referenced from:
                _helloworld_setup in helloworld.helloworld.40a883bh-cgu.0.rcgu.o
            "_gensym", referenced from:
                _helloworld_setup in helloworld.helloworld.40a883bh-cgu.0.rcgu.o
            "_class_new", referenced from:
                _helloworld_setup in helloworld.helloworld.40a883bh-cgu.0.rcgu.o
            "_pd_new", referenced from:
                helloworld::helloworld_new::h8642749205a325fe in helloworld.helloworld.40a883bh-cgu.0.rcgu.o
            "_post", referenced from:
                helloworld::helloworld_bang::h2e12cc0108f55cee in helloworld.helloworld.40a883bh-cgu.0.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)


error: aborting due to previous error

error: Could not compile `helloworld`.

To learn more, run the command again with --verbose.
```

**Comment by x37v** [comment]
fixed in 53f7abd81056576d3892e30ce0551800264f7720

gotta test again on linux now.

**Comment by x37v** [comment]
builds fine on my linux machine.
